centreon/src/Core/Connector/Application/UseCase/FindConnectors/FindConnectors.php

```diff
@@ -0,0 +1,175 @@
+                && ! $this->contact->hasTopologyRole(Contact::ROLE_CONFIGURATION_CONNECTORS_R)
+            ) {
+                $this->error(
+                    "User doesn't have sufficient rights to see connectors",
+                    ['user_id' => $this->contact->getId()]
+                );
+                $presenter->presentResponse(
+                    new ForbiddenResponse(ConnectorException::accessNotAllowed())
+                );
+
+                return;
+            }
+
+            $connectors = $this->readConnectorRepository->findByRequestParametersAndCommandTypes(
+                $this->requestParameters,
+                $this->retrieveConnectorTypesBasedOnContactRights(),
+            );
+
+            $commandIds = [];
+            foreach ($connectors as $connector)
+            {
+                array_push($commandIds, ...$connector->getcommandIds());
+            }
+
+            $commands = [];
+            if ($commandIds !== []) {
+                $commands = $this->readCommandRepository->findByIds($commandIds);
+            }
+
+            $presenter->presentResponse($this->createResponse($connectors, $commands));
+        } catch (RequestParametersTranslatorException $ex) {
+            $presenter->presentResponse(new ErrorResponse($ex->getMessage()));
+            $this->error($ex->getMessage(), ['trace' => $ex->getTraceAsString()]);
+        } catch (\Throwable $ex) {
+            $presenter->presentResponse(new ErrorResponse(ConnectorException::errorWhileSearching($ex)));
+            $this->error($ex->getMessage(), ['trace' => $ex->getTraceAsString()]);
+        }
+    }
+
+    /**
+     * @param Connector[] $connectors
+     * @param Command[] $commands
+     *
+     * @return FindConnectorsResponse
+     */
+    private function createResponse(array $connectors, array $commands): FindConnectorsResponse
+    {
+        $response = new FindConnectorsResponse();
+
+        foreach ($connectors as $connector) {
+            $connectorDto = new ConnectorDto();
+            $connectorDto->id = $connector->getId();
+            $connectorDto->name = $connector->getName();
+            $connectorDto->commandLine = $connector->getCommandLine();
+            $connectorDto->description = $connector->getDescription();
+            $connectorDto->isActivated = $connector->isActivated();
+            $connectorDto->commands = array_map(
+                fn(int $commandId) => [
+                    'id' => $commands[$commandId]->getId(),
+                    'name' => $commands[$commandId]->getName(),
+                    'type' => $commands[$commandId]->getType(),
+                ],
+                $connector->getCommandIds()
+            );
+            $response->connectors[] = $connectorDto;
+        }
+
+        return $response;
+    }
+
+    /**
+     * @return CommandType[]
+     */
+    private function retrieveConnectorTypesBasedOnContactRights(): array
```

Comment:
This method is 40 LoC and is 100% duplicate from the `FindCommands` use case.
Maybe we can do better like cerating a Domain Right class or something like that ?